### PR TITLE
feat: Upgrade value/action wiki articles with rake task

### PR DESCRIPTION
**Pull request by Mitcheljager**
This PR adds a rake task to upgrade wiki articles. This rake task is as ugly as it gets, but as it's a one time thing, I don't care. Verboseness over cleverness, and temporary cleverness over maintainability here.

Articles are changed to a new format;
- Description
- Snippet
- Properties
  - Return value
  - Parameters
    - Parameter details
    
Each article will now have a tiny snippet, this is super basic and only shows the value/action with it's default values. In some cases it doesn't make sense to use these in their isolated context, but it helps with copy pasting.

Previously the articles had a separate "return" section, but this is now under the "properties" section. This section lists both the return value and the parameters. The parameters are shown with a large details block, similar to before, with all their details given. In these blocks types are automatically given a link to their corresponding article if it exists. This is usually (always?) to constants like "HUD Reevaluation", "Color", "Button", etc.

The description is a little unique. Some articles have had their descriptions edited over time and I don't want to override those. If the description of the article is not the same as that in the yml file, we use the already existing description. If they are the same a new description is inserted with some new formatting.

Before | After
--- | ---
image |image
image | image
image | image
image | image
image | image
image | image
image | image

### CODE REVIEW

**Comment by netux** [comment]
> If the description of the article is not the same as that in the yml file, we use the already existing description

Did you also check for arguments getting changes?

File: lib/tasks/upgrade_wiki_articles.rake [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+#{ description_only == current["description"] ? "<big>#{ description_only }</big>" : description_only }
+
+## Snippet
+
+```
+#{ current["en-US"] }#{ current["args"].present? ? parameters_snippet : "" };
+```
+
+## Properties
```
Comment (netux) [inline lib/tasks/upgrade_wiki_articles.rake:67]:
```suggestion
## Arguments
```

Comment (netux) [reply lib/tasks/upgrade_wiki_articles.rake:67]: or

```suggestion
## Parameters
```

(matching what workshop.codes/editor calls them)

**Comment by Mitcheljager** [comment]
@netux Nope, those are intentionally overwritten.

File: lib/tasks/upgrade_wiki_articles.rake (hunk 1 above)
Comment (Mitcheljager) [reply lib/tasks/upgrade_wiki_articles.rake:67]: This is the general header for the properties of the action, which includes both the return value and parameters

Comment (netux) [reply lib/tasks/upgrade_wiki_articles.rake:67]: Ah I see. Didn't scroll further down 👍

Comment (Mitcheljager) [reply lib/tasks/upgrade_wiki_articles.rake:67]: I went with parameters over arguments. They are used interchangeably, but they don't actually mean the same thing. To put it in Javascript:

```
function thing(one, two) { ... } <- Parameters

thing(one, two) <- Arguments
```

The available options in the function are parameters, but you give arguments as values for those parameters.

**Pull request merged**